[Shivang Gupta] iP

Shivang211 opened the pull request:
# DukeIt
 
 > “Your mind is for having ideas, not holding them.” – David Allen [(source)](https://dansilvestre.com/productivity-quotes/)

DukeIt frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2122s2.github.io/website/schedule/week4/project.html)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it's **FREE**

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

 If you Java programmer, you can use it to practice Java too. Here's the `main` method
 
```java
public static void main(String[] args) throws DukeException, IOException {
        Duke duke = new Duke();
        duke.startDuke();
    }
```

Commit:
Add Gradle support

Shivang211 committed:
Add Level-1

Shivang211 committed:
Adding into list

Shivang211 committed:
Added mark and unmark

Shivang211 committed:
Add event, deadline and todo classes

Shivang211 committed:
Add level 4

Shivang211 committed:
added ui-testing

Shivang211 committed:
Added error checking

Shivang211 committed:
added delete

Shivang211 committed:
abstracting print

Shivang211 committed:
corrected ui-testing

Shivang211 committed:
new commit

Shivang211 committed:
add correct ui-testing

Shivang211 committed:
updated Duke.java

Shivang211 committed:
update gitignore

Shivang211 committed:
Add Initial Level-7

Shivang211 committed:
Add level-8

Shivang211 committed:
finish level-7

Shivang211 committed:
finish level-8

Shivang211 committed:
update gitignore

Shivang211 committed:
Merge branch 'branch-Level-7'

Shivang211 committed:
Merge branch 'branch-Level-8'

Shivang211 committed:
add more oop

Shivang211 committed:
fix project structure and allow intellij build

Shivang211 committed:
add packages

Shivang211 committed:
add junit

Shivang211 committed:
add javadoc

Shivang211 committed:
fix code to abide by the coding standard

Rdac0 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,12 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Parser {
```
Shouldn't you add JavaDoc comments?
I see very few.

Rdac0 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,139 @@
+                //toBeCompleted.isComplete();
+                displayIncompleteTask(toBeCompleted);
+            } else {
+                if (command.equals("event")) {
```
A bit of an arrowhead here. Is an extra indent necessary?

Rdac0 commented on src/main/java/exceptions/DukeUnknownCommandException.java:
```diff
@@ -0,0 +1,7 @@
+package exceptions;
+
+public class DukeUnknownCommandException extends Exception {
```
Couldn't this name be shorter? I see a lot of long names other than this too.

Rdac0 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,45 @@
+    }
+
+    public void startDuke() throws DukeException, IOException {
+        storage.loadData();
```
I think these lines could be more readable. Maybe you could add empty lines between logical units?

raynerljm replied:
I agree, it would help to make the code more readable

Rdac0 reviewed (commented):
A bit hard to read, could be better.

raynerljm commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,92 @@
+                String description = task.getDescription();
+                //task.isDone = false;
+                if (task instanceof ToDo) {
+                    text = "T | " + (task.getIsDone() ? "1" : "0") + " | " + description + '\n';
```
If you want to start a new line, the operator should be the first character

raynerljm commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,92 @@
+
+    public void loadData() throws IOException {
+        File file = new File(this.filePath);
+        boolean createDirectory = file.getParentFile().mkdirs();
```
boolean variables should start with (is, has, etc.)

raynerljm commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,92 @@
+            String[] commandWords = input.split(" \\| ", 4);
+            Task task = null;
+            switch (commandWords[0]) {
+                case "D":
```
Note the indentation for switch case, switch and case should be on the same indentation level

raynerljm commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,92 @@
+            sc.close();
+        }
+
+        public void display (String string){
```
There should not be a space between display and (String string)

raynerljm commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,139 @@
+                    LocalDate localDate = LocalDate.parse((descriptionAndTime[1].split(" ", 3))[1]);
+                    // System.out.println(descriptionAndTime[1].split(" ", 3)[1]);
+                    String deadlineTime = localDate.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
+//                DateTimeFormatter inputFormat = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
```
A lot of commented out code here. Is it needed?

raynerljm commented on src/main/java/tasks/TaskList.java:
```diff
@@ -0,0 +1,22 @@
+
+public class TaskList {
+
+    public static ArrayList<Task> dukeList = new ArrayList<>();
```
Variable names of collections should be plural

raynerljm reviewed (commented):
Overall really great! Just some minor coding standard notes to take note of. Checkstyles would really help a lot!

Shivang211 committed:
add level-9

Shivang211 committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-Add-Gradle

Shivang211 committed:
add assertions

Shivang211 committed:
Merge pull request #4 from Shivang211/branch-A-Assertions

add assertions

Shivang211 committed:
fix code quality

Shivang211 committed:
Merge pull request #5 from Shivang211/branch-A-CodeQuality

fix code quality

Shivang211 committed:
better coding practises

Shivang211 committed:
fix code

Shivang211 committed:
fix code

Shivang211 committed:
fix code

Shivang211 committed:
build.gradle : we are not using javafx and hence lack the dependencies for the same

Need to add the dependencies to be able to use JavaFX

Editing the build.gradle file and adding the relevant dependencies

We need to make our app more user friendly by including GUI

Shivang211 committed:
Set theme jekyll-theme-cayman

Shivang211 committed:
Set theme jekyll-theme-cayman

Shivang211 committed:
Set theme jekyll-theme-slate

Shivang211 committed:
Set theme jekyll-theme-slate

Shivang211 committed:
Set theme jekyll-theme-cayman

Shivang211 committed:
Update README.md

Shivang211 committed:
Update README.md

Shivang211 committed:
Update README.md

Shivang211 committed:
added BCD-extension

Shivang211 committed:
fix code

Shivang211 committed:
finish duke gui

Shivang211 committed:
add ui.png

Shivang211 committed:
fix smoke testing

Shivang211 committed:
add ui.png
